tests/foreman/cli/test_fact.py

```diff
@@ -100,22 +104,35 @@ def test_positive_update_client_facts_verify_imported_values(
         activation_keys=[module_activation_key.name],
     )
     assert result.status == 0, f'Failed to register host: {result.stderr}'
-    rhel_contenthost.execute('subscription-manager facts --update')
+
+    host = rhel_contenthost.nailgun_host
+    # gather ansible facts by running ansible roles on the host
+    task_id = host.play_ansible_roles()
+    module_target_sat.wait_for_tasks(
+        search_query=f'id = {task_id}',
+        poll_timeout=100,
+    )
+    task_details = module_target_sat.api.ForemanTask().search(query={'search': f'id = {task_id}'})
+    assert task_details[0].result == 'success'
+    assert rhel_contenthost.execute('subscription-manager facts --update').status == 0
     facts = module_target_sat.cli.Fact().list(
         options={'search': f'host={rhel_contenthost.hostname}'}, output_format='json'
     )
     facts_dict = {fact['fact']: fact['value'] for fact in facts}
     expected_values = {
         'net::interface::eth1::ipv4_address': ip,
-        'net::interface::eth1::mac_address': mac_address.lower(),
         'network::fqdn': rhel_contenthost.hostname,
         'lscpu::architecture': rhel_contenthost.arch,
+        'ansible_distribution_major_version': str(rhel_contenthost.os_version.major),
+        'ansible_fqdn': rhel_contenthost.hostname,
     }
     for fact, expected_value in expected_values.items():
         actual_value = facts_dict.get(fact)
         assert (
             actual_value == expected_value
         ), f'Assertion failed: {fact} (expected: {expected_value}, actual: {actual_value})'
+    if not is_open('SAT-27056'):
+        assert facts_dict.get('net::interface::eth1::mac_address') == mac_address.lower()
```

Comment:
just to optimize this assertion, you could also include this into expected_values dict, so we don't assert it explicitely even if issues is fixed in future
```suggestion
    if not is_open('SAT-27056'):
        expected_values['net::interface::eth1::mac_address'] = mac_address.lower()
```

Reply:
Sure.